atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/qos/ThriftQueryWeighers.java

```diff
@@ -42,40 +42,56 @@
     private static final Logger log = LoggerFactory.getLogger(CassandraClient.class);
 
     @VisibleForTesting
+    private static final int ESTIMATED_NUM_BYTES_PER_ROW = 100;
     static final QueryWeight DEFAULT_ESTIMATED_WEIGHT = ImmutableQueryWeight.builder()
-            .numBytes(100)
+            .numBytes(ESTIMATED_NUM_BYTES_PER_ROW)
             .numDistinctRows(1)
             .timeTakenNanos(TimeUnit.MILLISECONDS.toNanos(2))
             .build();
 
     private ThriftQueryWeighers() { }
 
-    public static final QosClient.QueryWeigher<Map<ByteBuffer, List<ColumnOrSuperColumn>>> MULTIGET_SLICE =
-            readWeigher(ThriftObjectSizeUtils::getApproximateReadByteCount, Map::size);
+    static QosClient.QueryWeigher<Map<ByteBuffer, List<ColumnOrSuperColumn>>> multigetSlice(
+            int numberOfQueriedRows) {
+        return readWeigher(ThriftObjectSizeUtils::getApproximateSizeOfColsByKey, Map::size, numberOfQueriedRows);
+    }
 
-    public static final QosClient.QueryWeigher<List<KeySlice>> GET_RANGE_SLICES =
-            readWeigher(ThriftObjectSizeUtils::getApproximateReadByteCount, List::size);
+    static QosClient.QueryWeigher<List<KeySlice>> getRangeSlices(
+            int numberOfQueriedRows) {
+        return readWeigher(ThriftObjectSizeUtils::getApproximateSizeOfKeySlices, List::size, numberOfQueriedRows);
+    }
 
-    public static final QosClient.QueryWeigher<ColumnOrSuperColumn> GET =
-            readWeigher(ThriftObjectSizeUtils::getColumnOrSuperColumnSize, ignored -> 1);
+    static final QosClient.QueryWeigher<ColumnOrSuperColumn> GET =
+            readWeigher(ThriftObjectSizeUtils::getColumnOrSuperColumnSize, ignored -> 1, 1);
 
-    public static final QosClient.QueryWeigher<CqlResult> EXECUTE_CQL3_QUERY =
+    static final QosClient.QueryWeigher<CqlResult> EXECUTE_CQL3_QUERY =
             // TODO(nziebart): we need to inspect the schema to see how many rows there are - a CQL row is NOT a
             // partition. rows here will depend on the type of query executed in CqlExecutor: either (column, ts) pairs,
             // or (key, column, ts) triplets
-            readWeigher(ThriftObjectSizeUtils::getCqlResultSize, ignored -> 1);
+            readWeigher(ThriftObjectSizeUtils::getCqlResultSize, ignored -> 1, 1);
 
-    public static QosClient.QueryWeigher<Void> batchMutate(
+    static QosClient.QueryWeigher<Void> batchMutate(
             Map<ByteBuffer, Map<String, List<Mutation>>> mutationMap) {
         long numRows = mutationMap.size();
-        return writeWeigher(numRows, () -> ThriftObjectSizeUtils.getApproximateWriteByteCount(mutationMap));
+        return writeWeigher(numRows, () -> ThriftObjectSizeUtils.getApproximateSizeOfMutationMap(mutationMap));
+    }
+
+    public static QosClient.QueryWeigher<CASResult> cas(List<Column> updates) {
```

Comment:
did you mean to add this back?

Reply:
This may have been an artifact of the commit-shuffling

Reply:
...but it seems to be gone again now, so we're good 👍